neuraxle/base.py

```diff
@@ -2179,37 +2190,51 @@ def teardown(self) -> 'BaseStep':
 
         return self
 
-    def apply(self, method_name: str, *kargs, **kwargs) -> Union[Dict, Iterable]:
+    def apply(self, method_name: str, step_name=None, *kargs, **kwargs) -> Dict:
         """
         Apply the method name to the pipeline step and all of its children.
 
         :param method_name: method name that need to be called on all steps
+        :param step_name: current pipeline step name
         :param kargs: any additional arguments to be passed to the method
         :param kwargs: any additional positional arguments to be passed to the method
         :return: accumulated results
         :rtype: Union[Dict, Iterable]
         """
-        results = BaseStep.apply(self, method_name, *kargs, **kwargs)
+        results = BaseStep.apply(self, method_name, step_name=step_name, *kargs, **kwargs)
+
+        if step_name is not None:
+            step_name = "{}__{}".format(step_name, self.name)
```

Comment:
Isn't `__` already a constant elsewhere that you can use and that is defined at the top of the base.py file?